core/src/main/scala/kafka/log/LogCleanerManager.scala

```diff
@@ -595,8 +595,8 @@ private[log] object LogCleanerManager extends Logging {
     // may be cleaned
     val firstUncleanableDirtyOffset: Long = Seq(
 
-      // we do not clean beyond the first unstable offset
-      log.firstUnstableOffset,
+      // we do not clean beyond the last stable offset
```

Comment:
This is an existing issue. But could we update the comment in line 593 to include last stable offset too?

Reply:
Makes sense. I've updated it with a better explanation of what we bound by.